app/src/main/java/com/kabouzeid/gramophone/appwidgets/AppWidgetCard.java

```diff
@@ -0,0 +1,228 @@
+        final int widgetImageSize = service.getResources().getDimensionPixelSize(R.dimen.app_widget_small_image_size);
+        service.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if (target != null) {
+                    Glide.clear(target);
+                }
+                target = SongGlideRequest.Builder.from(Glide.with(appContext), song)
+                        .checkIgnoreMediaStore(appContext)
+                        .generatePalette(service).build()
+                        .into(new SimpleTarget<BitmapPaletteWrapper>(widgetImageSize, widgetImageSize) {
+                            @Override
+                            public void onResourceReady(BitmapPaletteWrapper resource, GlideAnimation<? super BitmapPaletteWrapper> glideAnimation) {
+                                Palette palette = resource.getPalette();
+                                update(resource.getBitmap(), palette.getVibrantColor(palette.getMutedColor(MaterialValueHelper.getSecondaryTextColor(appContext, true))));
+                            }
+
+                            @Override
+                            public void onLoadFailed(Exception e, Drawable errorDrawable) {
+                                super.onLoadFailed(e, errorDrawable);
+                                update(null, MaterialValueHelper.getSecondaryTextColor(appContext, true));
+                            }
+
+                            private void update(@Nullable Bitmap bitmap, int color) {
+                                appWidgetView.setViewVisibility(R.id.image, View.VISIBLE);
+
+                                // Set correct drawable for pause state
+                                int playPauseRes = isPlaying ? R.drawable.ic_pause_white_24dp : R.drawable.ic_play_arrow_white_24dp;
+                                appWidgetView.setImageViewBitmap(R.id.button_toggle_play_pause, createBitmap(Util.getTintedVectorDrawable(service, playPauseRes, color), 1f));
+
+                                // Set prev/next button drawables
+                                appWidgetView.setImageViewBitmap(R.id.button_next, createBitmap(Util.getTintedVectorDrawable(service, R.drawable.ic_skip_next_white_24dp, color), 1f));
+                                appWidgetView.setImageViewBitmap(R.id.button_prev, createBitmap(Util.getTintedVectorDrawable(service, R.drawable.ic_skip_previous_white_24dp, color), 1f));
+
+                                Drawable image;
+
+                                if (bitmap == null) {
+                                    image = appContext.getResources().getDrawable(R.drawable.default_album_art);
+                                } else {
+                                    image = new BitmapDrawable(bitmap);
+                                }
+
+                                float radius = appContext.getResources().getDimension(R.dimen.app_widget_card_radius);
```

Comment:
I'm wondering if you could just use a `CardView` in the layout so you don't actually need to do this explicitely.